---
Title: vagrant box updateでエラー「An error occurred while downloading the remote file」
Category:
- vagrant
Date: 2018-03-19T10:33:53+09:00
URL: https://blog.turai.work/entry/20180319/1521423233
EditURL: https://blog.hatena.ne.jp/thr3a/thr3a.hatenablog.com/atom/entry/17391345971627117904
---

ふと、[bento/ubuntu-16.04](https://app.vagrantup.com/bento/boxes/ubuntu-16.04)のVagrant boxをアップデートしてやろう思って

```
vagrant box update
```

したが、エラーになって何回やってもできない

```
An error occurred while downloading the remote file. The error 
message, if any, is reproduced below. Please fix this error and try 
again.
```

# 原因

boxのダウンロード元URLが変わっているのが原因。

boxのURLはMacだと` ~/.vagrant.d/boxes/BOX名/metadata_url`に記載されている。今回だと

```
$ cat ~/.vagrant.d/boxes/bento-VAGRANTSLASH-ubuntu-16.04/metadata_url 
https://atlas.hashicorp.com/bento/ubuntu-16.04
```

になっていたが、

```
https://vagrantcloud.com/bento/ubuntu-16.04-i386
```

に変えてあげる必要がある。

そもそもbento/ubuntu-16.04については　ubuntu-16.04-i386って名前に変わってたってのも問題だったが、、、

とりあえず解決したので
